src/nvim/api/buffer.c

```diff
@@ -179,6 +179,11 @@ void buffer_set_line_slice(Buffer buffer,
     return;
   }
 
+  if (!inbounds(buf, start)) {
+    api_set_error(err, Validation, _("Index out of bounds"));
+    return;
```

Comment:
Yes, but I daresay that the latter invocation is significantly less clear! Furthermore, since atomic atomic operations are currently unsupported, by the time that the plugin calculated `-n-1`, the actual line count could have changed. It seems dangerous to rely on this side effect.